web/pdf_viewer.js

```diff
@@ -135,833 +133,825 @@ var PDFViewer = (function pdfViewer() {
-          // since aligning the bottom of the intended page with the
-          // top of the window is rarely helpful).
-          x = x !== null ? x : 0;
-          y = y !== null ? y : pageHeight;
+      let hPadding = (this.isInPresentationMode || this.removePageBorders) ?
+        0 : SCROLLBAR_PADDING;
+      let vPadding = (this.isInPresentationMode || this.removePageBorders) ?
+        0 : VERTICAL_PADDING;
+      let pageWidthScale = (this.container.clientWidth - hPadding) /
+                           currentPage.width * currentPage.scale;
+      let pageHeightScale = (this.container.clientHeight - vPadding) /
+                            currentPage.height * currentPage.scale;
+      switch (value) {
+        case 'page-actual':
+          scale = 1;
           break;
-        case 'Fit':
-        case 'FitB':
-          scale = 'page-fit';
+        case 'page-width':
+          scale = pageWidthScale;
           break;
-        case 'FitH':
-        case 'FitBH':
-          y = dest[2];
-          scale = 'page-width';
-          // According to the PDF spec, section 12.3.2.2, a `null` value in the
-          // parameter should maintain the position relative to the new page.
-          if (y === null && this._location) {
-            x = this._location.left;
-            y = this._location.top;
-          }
+        case 'page-height':
+          scale = pageHeightScale;
           break;
-        case 'FitV':
-        case 'FitBV':
-          x = dest[2];
-          width = pageWidth;
-          height = pageHeight;
-          scale = 'page-height';
+        case 'page-fit':
+          scale = Math.min(pageWidthScale, pageHeightScale);
           break;
-        case 'FitR':
-          x = dest[2];
-          y = dest[3];
-          width = dest[4] - x;
-          height = dest[5] - y;
-          var hPadding = this.removePageBorders ? 0 : SCROLLBAR_PADDING;
-          var vPadding = this.removePageBorders ? 0 : VERTICAL_PADDING;
-
-          widthScale = (this.container.clientWidth - hPadding) /
-            width / CSS_UNITS;
-          heightScale = (this.container.clientHeight - vPadding) /
-            height / CSS_UNITS;
-          scale = Math.min(Math.abs(widthScale), Math.abs(heightScale));
+        case 'auto':
+          let isLandscape = (currentPage.width > currentPage.height);
+          // For pages in landscape mode, fit the page height to the viewer
+          // *unless* the page would thus become too wide to fit horizontally.
+          let horizontalScale = isLandscape ?
+            Math.min(pageHeightScale, pageWidthScale) : pageWidthScale;
+          scale = Math.min(MAX_AUTO_SCALE, horizontalScale);
           break;
         default:
-          console.error('PDFViewer_scrollPageIntoView: \'' + dest[1].name +
-                        '\' is not a valid destination type.');
+          console.error(
+            'PDFViewer._setScale: "${value}" is an unknown zoom value.');
```

Comment:
This whole string should be in backticks instead of regular quotes to actually evaluate the template literal "${value}".